File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/trackedsections/TrackedSectionsSubcomponent.java
package com.tevinjeffrey.rutgersct.ui.trackedsections;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@Subcomponent
public interface TrackedSectionsSubcomponent
    extends AndroidInjector<TrackedSectionsFragment> {

  void inject(TrackedSectionsPresenterImpl trackedSectionsPresenter);

  @Subcomponent.Builder
  abstract class Builder extends AndroidInjector.Builder<TrackedSectionsFragment> {
    @Override public abstract TrackedSectionsSubcomponent build();
  }
}

File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/trackedsections/TrackedSectionsSubcomponent.kt
package com.tevinjeffrey.rutgersct.ui.trackedsections

import dagger.Subcomponent
import dagger.android.AndroidInjector

@Subcomponent
interface TrackedSectionsSubcomponent : AndroidInjector<TrackedSectionsFragment> {

  fun inject(trackedSectionsPresenter: TrackedSectionsViewModel)

  @Subcomponent.Builder
  abstract class Builder : AndroidInjector.Builder<TrackedSectionsFragment>() {
    abstract override fun build(): TrackedSectionsSubcomponent
  }
}
